actionpack/lib/action_dispatch/http/request.rb

```diff
@@ -358,6 +358,7 @@ def body_stream # :nodoc:
 
     def reset_session
       session.destroy
+      controller_instance&.reset_csrf_token(self)
```

Comment:
I don't know if this is the right place to be doing this or not.  Putting this on `ActionController` instead of here would mean that it's possible to reset the session and not the cookie, so this felt safer.

It might be nice if the session could refer back to the token store, then we could do this in destroy, but it doesn't feel like that is something that Session should know about.

I have no idea how to write a unit test for this.  If we need one, I'm going to need guidance.